artifactory_test.go

```diff
@@ -4511,3 +4512,56 @@ func uploadWithSpecificServerAndVerify(t *testing.T, cli *tests.JfrogCli, server
 	assert.Len(t, searchItemsInArtifactory(t, tests.SearchRepo1ByInSuffix), expectedResults)
 	return nil
 }
+
+func TestPermissionTargets(t *testing.T) {
+	initArtifactoryTest(t)
+	servicesManager, err := utils.CreateServiceManager(artifactoryDetails, false)
+	if err != nil {
+		assert.NoError(t, err)
+		return
+	}
+	templatePath := filepath.Join(tests.GetTestResourcesPath(), "permissiontarget", "template")
+
+	// Create permission target on specific repo.
+	assert.NoError(t, artifactoryCli.Exec("ptc", templatePath, createPermissionTargetsTemplateVars(tests.RtRepo1)))
+	err = getAndAssertExpectedPermissionTarget(t, servicesManager, tests.RtRepo1)
+	if err != nil {
+		return
+	}
+
+	defer cleanPermissionTarget()
```

Comment:
Since we delete the permission target at the end of this test function, I suggest we modify this defer as follows:
```
		defer func() {
                 if !permissionDeleted {
                     cleanPermissionTarget()
                }
 		}()
```